src/ir/datalog/attribute.h

```diff
@@ -24,34 +24,41 @@
 
 namespace raksha::ir::datalog {
 
-inline constexpr absl::string_view kStringAttributePayloadName =
-    "StringAttributePayload";
-inline constexpr absl::string_view kNumberAttributePayloadName =
-    "NumberAttributePayload";
+using AttributePayload = Adt;
 
-class AttributePayload : public Adt {
+class Attribute
+    : public Record<Symbol /*attr_name*/, AttributePayload /*attr_payload*/> {
  public:
-  using Adt::Adt;
-};
+  explicit Attribute(absl::string_view name, AttributePayload payload)
+      : Record(Symbol(name), std::move(payload)) {}
 
-class StringAttributePayload : public AttributePayload {
- public:
-  StringAttributePayload(Symbol symbol)
-      : AttributePayload(kStringAttributePayloadName) {
-    arguments_.push_back(std::make_unique<Symbol>(symbol));
-  }
-};
+  class String : public AttributePayload {
+   public:
+    explicit String(Symbol symbol)
+        : AttributePayload(kStringAttributePayloadName) {
+      arguments_.push_back(std::make_unique<Symbol>(symbol));
```

Comment:
Should probably `std::move(symbol)` here.

Reply:
done.